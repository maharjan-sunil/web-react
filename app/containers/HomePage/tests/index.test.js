/**
 * Test the HomePage
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory, MemoryRouter } from 'react-router-dom';

import { HomePage } from '../index';
import configureStore from '../../../configureStore';

describe('<HomePage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <MemoryRouter>
          <IntlProvider locale="en">
            <HomePage loading={false} error={false} repos={[]} />
          </IntlProvider>
        </MemoryRouter>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
