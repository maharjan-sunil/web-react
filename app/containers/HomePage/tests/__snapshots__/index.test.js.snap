// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<nav
  class="Navbar-sc-1o0l5dr-0 StyledNavbar-sc-13g77u5-0 eoxpNL navbar navbar-toggleable-lg"
  role="navigation"
>
  <a
    class="StyledNavbar__StyleNavbarBrand-sc-13g77u5-1 gAQlWc navbar-brand"
  >
    <img
      alt="Wallbee Logo"
      class="Img-sc-10y4zbe-0 fFvzSV"
      src="IMAGE_MOCK"
    />
  </a>
  <button
    aria-controls="navbarSupportedContent"
    class="navbar-toggler"
    type="button"
  >
    <span
      class="navbar-toggler-icon"
    />
  </button>
  <div
    class="Collapse-wjso76-0 bSLYTt collapse navbar-collapse"
    id="navbarSupportedContent"
  >
    <ul
      class="Nav-ggw7qv-0 mr-auto StyledNavbar__MainNav-sc-13g77u5-2 lhlDeI navbar-nav Ul-sctud2-0 iJDZqn"
      role="navigation"
      tag="nav"
      toggleable="false"
    >
      <li
        class="StyledNavbar__StyledNavItem-sc-13g77u5-3 ceCYsr nav-item"
      >
        <a
          class="nav-link A-sc-70cat-0 gsCFrI"
          href="/"
          tag="a"
          theme="[object Object]"
        >
          <span>
            Home
          </span>
        </a>
      </li>
      <li
        class="StyledNavbar__StyledNavItem-sc-13g77u5-3 ceCYsr nav-item"
      >
        <a
          class="nav-link A-sc-70cat-0 gsCFrI"
          href="/logs"
          tag="a"
          theme="[object Object]"
        >
          <span>
            Log
          </span>
        </a>
      </li>
      <li
        class="StyledNavbar__StyledNavItem-sc-13g77u5-3 ceCYsr nav-item"
      >
        <a
          class="nav-link A-sc-70cat-0 gsCFrI"
          href="/sites"
          tag="a"
          theme="[object Object]"
        >
          <span>
            Site
          </span>
        </a>
      </li>
      <li
        class="StyledNavbar__StyledNavItem-sc-13g77u5-3 ceCYsr nav-item"
      >
        <a
          class="nav-link A-sc-70cat-0 gsCFrI"
          href="/site-accounts"
          tag="a"
          theme="[object Object]"
        >
          <span>
            Site Account
          </span>
        </a>
      </li>
      <li
        class="StyledNavbar__StyledNavItem-sc-13g77u5-3 ceCYsr nav-item"
      >
        <a
          class="nav-link A-sc-70cat-0 gsCFrI"
          href="/scans"
          tag="a"
          theme="[object Object]"
        >
          <span>
            Scan
          </span>
        </a>
      </li>
      <li
        class="StyledNavbar__StyledNavItem-sc-13g77u5-3 ceCYsr nav-item"
      >
        <a
          class="nav-link A-sc-70cat-0 gsCFrI"
          href="/data-connections"
          tag="a"
          theme="[object Object]"
        >
          <span>
            Data Connection
          </span>
        </a>
      </li>
      <li
        class="StyledNavbar__StyledNavItem-sc-13g77u5-3 ceCYsr nav-item"
      >
        <a
          class="nav-link A-sc-70cat-0 gsCFrI"
          href="/carriers"
          tag="a"
          theme="[object Object]"
        >
          <span>
            Carrier
          </span>
        </a>
      </li>
    </ul>
  </div>
  <ul
    class="Nav-ggw7qv-0 StyledNavbar__MainNavIcon-sc-13g77u5-8 ejRcrn nav Ul-sctud2-0 iJDZqn"
    role="navigation"
    tag="nav"
    toggleable="false"
  >
    <li
      class="StyledNavbar__StyledNavItemIcon-sc-13g77u5-7 bRFWWk nav-item"
    >
      <svg
        aria-hidden="true"
        class="StyledIconBase-ea9ulj-0 iLHuDN"
        fill="currentColor"
        focusable="false"
        height="12"
        viewBox="0 0 384 512"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M202.021 0C122.202 0 70.503 32.703 29.914 91.026c-7.363 10.58-5.093 25.086 5.178 32.874l43.138 32.709c10.373 7.865 25.132 6.026 33.253-4.148 25.049-31.381 43.63-49.449 82.757-49.449 30.764 0 68.816 19.799 68.816 49.631 0 22.552-18.617 34.134-48.993 51.164-35.423 19.86-82.299 44.576-82.299 106.405V320c0 13.255 10.745 24 24 24h72.471c13.255 0 24-10.745 24-24v-5.773c0-42.86 125.268-44.645 125.268-160.627C377.504 66.256 286.902 0 202.021 0zM192 373.459c-38.196 0-69.271 31.075-69.271 69.271 0 38.195 31.075 69.27 69.271 69.27s69.271-31.075 69.271-69.271-31.075-69.27-69.271-69.27z"
          fill="currentColor"
        />
      </svg>
    </li>
    <li
      class="StyledNavbar__StyledNavItemIcon-sc-13g77u5-7 bRFWWk nav-item"
    >
      <svg
        aria-hidden="true"
        class="StyledIconBase-ea9ulj-0 iLHuDN"
        fill="currentColor"
        focusable="false"
        height="12"
        viewBox="0 0 448 512"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M439.39 362.29c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71zM67.53 368c21.22-27.97 44.42-74.33 44.53-159.42 0-.2-.06-.38-.06-.58 0-61.86 50.14-112 112-112s112 50.14 112 112c0 .2-.06.38-.06.58.11 85.1 23.31 131.46 44.53 159.42H67.53zM224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64z"
          fill="currentColor"
        />
      </svg>
    </li>
    <li
      class="StyledNavbar__StyledNavItemIcon-sc-13g77u5-7 bRFWWk nav-item"
    >
      <svg
        aria-hidden="true"
        class="StyledIconBase-ea9ulj-0 iLHuDN"
        fill="currentColor"
        focusable="false"
        height="12"
        viewBox="0 0 512 512"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
          fill="currentColor"
        />
      </svg>
    </li>
    <li
      class="StyledNavbar__StyledNavItemIcon-sc-13g77u5-7 bRFWWk nav-item"
    >
      <svg
        aria-hidden="true"
        class="StyledIconBase-ea9ulj-0 iLHuDN"
        fill="currentColor"
        focusable="false"
        height="12"
        viewBox="0 0 448 512"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
          fill="currentColor"
        />
      </svg>
    </li>
  </ul>
</nav>
`;
